exporters/otlp/src/otlp_grpc_metric_exporter.cc

```diff
@@ -60,35 +66,85 @@ opentelemetry::sdk::common::ExportResult OtlpGrpcMetricExporter::Export(
     return sdk::common::ExportResult::kSuccess;
   }
 
-  proto::collector::metrics::v1::ExportMetricsServiceRequest request;
-  OtlpMetricUtils::PopulateRequest(data, &request);
+  google::protobuf::ArenaOptions arena_options;
+  // It's easy to allocate datas larger than 1024 when we populate basic resource and attributes
+  arena_options.initial_block_size = 1024;
+  // When in batch mode, it's easy to export a large number of spans at once, we can alloc a lager
+  // block to reduce memory fragments.
+  arena_options.max_block_size = 65536;
+  std::unique_ptr<google::protobuf::Arena> arena{new google::protobuf::Arena{arena_options}};
 
-  auto context = OtlpGrpcClient::MakeClientContext(options_);
-  proto::collector::metrics::v1::ExportMetricsServiceResponse response;
+  proto::collector::metrics::v1::ExportMetricsServiceRequest *request =
+      google::protobuf::Arena::Create<proto::collector::metrics::v1::ExportMetricsServiceRequest>(
+          arena.get());
+  OtlpMetricUtils::PopulateRequest(data, request);
 
-  grpc::Status status = OtlpGrpcClient::DelegateExport(metrics_service_stub_.get(), context.get(),
-                                                       request, &response);
+  auto context = OtlpGrpcClient::MakeClientContext(options_);
+  proto::collector::metrics::v1::ExportMetricsServiceResponse *response =
+      google::protobuf::Arena::Create<proto::collector::metrics::v1::ExportMetricsServiceResponse>(
+          arena.get());
 
-  if (!status.ok())
+#ifdef ENABLE_ASYNC_EXPORT
+  if (options_.max_concurrent_requests > 1)
   {
-    OTEL_INTERNAL_LOG_ERROR(
-        "[OTLP METRIC GRPC Exporter] Export() failed: " << status.error_message());
-    return sdk::common::ExportResult::kFailure;
+    return client_->DelegateAsyncExport(
+        options_, metrics_service_stub_.get(), std::move(context), std::move(arena),
+        std::move(*request),
+        [](opentelemetry::sdk::common::ExportResult result,
+           std::unique_ptr<google::protobuf::Arena> &&,
+           const proto::collector::metrics::v1::ExportMetricsServiceRequest &request,
+           proto::collector::metrics::v1::ExportMetricsServiceResponse *) {
+          if (result != opentelemetry::sdk::common::ExportResult::kSuccess)
+          {
+            OTEL_INTERNAL_LOG_ERROR("[OTLP HTTP Client] ERROR: Export "
+                                    << request.resource_metrics_size()
+                                    << " metric(s) error: " << static_cast<int>(result));
+          }
+          else
+          {
+            OTEL_INTERNAL_LOG_DEBUG("[OTLP HTTP Client] Export " << request.resource_metrics_size()
```

Comment:
OTLP METRIC GRPC Exporter

Please adjust all the logging for the HTTP exporters as well, they use "OTLP HTTP Client" instead of "OTLP TRACE/METRIC/LOGS HTTP Exporter".

Reply:
Thanks and one.